util/topgen/merge.py

```diff
@@ -408,18 +408,35 @@ def xbar_cross_node(node_name, device_xbar, xbars, visited=[]):
 
     visited.pop()
 
-    # TODO: simplify the result? if contiguous, combine into one?
-    return simplify_addr(result, device_xbar)
+    return result
+
+
+def simplify_xbar(xbars):
```

Comment:
if i understand this correctly...this means (using your example), if current entry is 0x4000_0000, 0x21000, and next is 0x4008_0000, we will use 0x4000_0000 ~ 0x4008_0000 during the decode, meaning anything from 0x21000~0x80000 will not actually fall into the default slave, and instead, it is the peripheral that has to correctly handle that, is that right...?

Reply:
Yes, any unmapped address request will be returned with error at the lower-tiered crossbar.

Reply:
does that also apply to peripheral devices? For example...if some reason we have a peripheral that had a size_byte that was not power of 2 aligned?

Reply:
It is limited to reduce the computation. Applying to the leaf devices doesn't help in my opinion as the address and size are given by users.

Reply:
ah okay that sounds good. But does that mean if a peripheral device doesn't provide nice numbers we will error out?